tests/tox_env/python/test_python_api.py

```diff
@@ -131,3 +132,18 @@ def test_base_python_env_conflict_show_conf(tox_project: ToxProjectCreator, igno
             f" base python py{py_ver_next}'{',' if comma_in_exc else ''})\n"
         )
     result.assert_out_err(out, "")
+
+
+def test_path_from_set_env(tox_project: ToxProjectCreator) -> None:
+    tox_ini = """
+        [testenv]
+        package = skip
+        set_env =
+            PATH=/dev/null
```

Comment:
I'm not sure this is a valid use case. By definition of tox path is where we find the executables used in commands; if you set it to null means no commands are valid. I'm willing to make this use case just not supported.

Reply:
> I'm willing to make this use case just not supported.

Whether intended or not, this did work in tox 3.

That is the PATH from the set_env (prefixed by the virtual env's bindir) is used to find the executables used in the commands executed within that environment (as well as affecting the PATH environment variable seen by any of those commands.)
However, when *constructing* the virtual environment, tox 3 searches `os.environ["PATH"]` (unmodified by any `set_env` setting) to find the python executable.  Thereafter, the prefixing of any PATH with the virtual env's bindir ensures that python, as well as any other commands installed in the virtual env can be found regardless of how badly set_env mangles PATH.

(I do want it to mean "no commands, except those directly installed in the virtual env, are valid".)

----

This is exactly the use case I described in issue #2691.  Where I also asked, "Is this expected?".
You [replied](https://github.com/tox-dev/tox/issues/2691#issuecomment-1347224886), "PR to fix this is welcome", which I took to mean that this is in fact a valid use case.
If it's not, say so, and I will find some other way forward.

Reply:
> Whether intended or not, this did work in tox 3.

True, but that could have been by chance instead by design. 

> This is exactly the use case I described in issue #2691. Where I also asked, "Is this expected?".
> You [replied](https://github.com/tox-dev/tox/issues/2691#issuecomment-1347224886), "PR to fix this is welcome", which I took to mean that this is in fact a valid use case.
> If it's not, say so, and I will find some other way forward

Sorry for that, did not think through initially of its implications.

Reply:
I think your request is valid and sane; I'm just not sure this is the right solution.